blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java

```diff
@@ -75,6 +72,10 @@
     private TimeValue timeout;
     private static int MAX_CONTENT_LENGTH = 1048576; // 1 MB
 
+    private ServerBootstrap bootstrap;
+    private ChannelFactory channelFactory;
+    private ChannelGroup allOpenChannels = new DefaultChannelGroup("allOpenChannels");
```

Comment:
True. I had thought about adding a stage to the pipeline (in `MetricsHttpServerPipelineFactory`'s `getPipeline` method) to collect all channels that get created into `allOpenChannels`. Then when we shut it down, we could make sure all channels get closed, and call `releaseExternalResources` on the factory as well. However, I'm not entirely sure about that right now. Keeping track of all channels that get created, over an extended period of time, could consume a lot of memory needlessly. It would depend on whether `ChannelGroup` is smart enough to *remove* closed channels from itself automatically, which I don't know yet. I'll remove the extraneous fields for now.